recipes/z3/all/conanfile.py

```diff
@@ -21,109 +23,104 @@ class Z3Conan(ConanFile):
         "shared": [True, False],
         "fPIC": [True, False],
         "multithreaded": [True, False],
-        "multiprecision": ["internal", "gmp", "mpir"]
+        "use_gmp": [True, False]
```

Comment:
I've read your rational about this change, but do you know that there is set_property in CMakeDeps to avoid patching upstream CMakeLists? I think we can keep this option without any maintenance burden with some smart set_property.

Reply:
I see what you means. We can pretend that the target name of the MPIR library is `GMP::GMP`. I will give a try. Thanks for the suggestions.